android/app/src/androidTest/java/com/trustwallet/core/app/utils/TestHDWallet.kt

```diff
@@ -38,13 +38,13 @@ class TestHDWallet {
 
     @Test
     fun testCreateFromMnemonicInvalid() {
-        var exception = false
+        var exception = ""
         try {
             val hd = HDWallet("THIS IS AN INVALID MNEMONIC", "")
         } catch (ex: Exception) {
-            exception = true
+            exception = ex::class.simpleName
         }
-        assertTrue(exception)
+        assertEquals(exception, "InvalidParameterException")
```

Comment:
@vcoolish what's the better way to check an error?

Reply:
```kotlin
    @Test
    fun testCreateFromMnemonicInvalid() {
        try {
            HDWallet("THIS IS AN INVALID MNEMONIC", "")
            fail("Assert exception")
        } catch (ex: Exception) {
            assertTrue(ex is InvalidParameterException)
        }
    }
```

Reply:
I tried the more elegant assertFailWith (no try/catch needed), but it requires new dependency kotlin.test, which I don't want to add now
https://github.com/trustwallet/wallet-core/pull/1560
So I'll just stick with your suggestion.